flytekit/core/type_engine.py

```diff
@@ -598,13 +613,20 @@ def to_python_value(self, ctx: FlyteContext, lv: Literal, expected_python_type:
                 f"{expected_python_type} is not of type @dataclass, only Dataclasses are supported for "
                 f"user defined datatypes in Flytekit"
             )
-        if not issubclass(expected_python_type, DataClassJsonMixin):
+        if not issubclass(expected_python_type, DataClassJsonMixin) and not issubclass(
+            expected_python_type, DataClassJSONMixin
+        ):
             raise TypeTransformerFailedError(
-                f"Dataclass {expected_python_type} should be decorated with @dataclass_json to be "
+                f"Dataclass {expected_python_type} should be decorated with @dataclass_json or mixin with DataClassJSONMixin to be "
                 f"serialized correctly"
             )
         json_str = _json_format.MessageToJson(lv.scalar.generic)
-        dc = cast(DataClassJsonMixin, expected_python_type).from_json(json_str)
+
+        if issubclass(expected_python_type, DataClassJsonMixin):
+            dc = cast(DataClassJsonMixin, expected_python_type).from_json(json_str)  # type: ignore
+        else:
+            dc = cast(DataClassJSONMixin, expected_python_type).from_json(json_str)  # type: ignore
```

Comment:
```suggestion
        dc = cast(DataClassJSONMixin, expected_python_type).from_json(json_str)  # type: ignore
```

Reply:
Hi @pingsutw, some problems to ask.
we keep 
 dc = cast(DataClassJSONMixin, expected_python_type).from_json(json_str)  with mashumaro,
but we use 
json_str = cast(DataClassJsonMixin, python_val).to_json()  # type: ignore with dataclass json

Should we be consistent through the changes?